datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetAvroConverter.java

```diff
@@ -0,0 +1,343 @@
+                  if (array != null) {
+                    ListIterator<GenericRecord> it = array.listIterator();
+                    while (it.hasNext()) {
+                      GenericRecord nestedObjectRecord = it.next();
+                      GenericRecord parquetFormatRecord = generateParquetStructuredAvroData(generateParquetStructuredAvroSchema(avroTypeSchema.getElementType()), nestedObjectRecord);
+                      nestedObjectArray.add(parquetFormatRecord);
+                    }
+                  }
+                  record.put(fieldName.toLowerCase(), nestedObjectArray);
+                } else {
+                  record.put(fieldName.toLowerCase(), avroRecord.get(fieldName));
+                }
+              } else if (avroTypeName.equalsIgnoreCase(AVRO_LONG)) {
+                // For timestamp fields we need to check for logical type in avroschema, to convert to string in parquet
+                if (avroTypeSchema.getLogicalType() != null &&
+                        avroTypeSchema.getLogicalType().getName().equalsIgnoreCase("timestamp-millis")) {
+                  String result = null;
+                  if (avroRecord.get(fieldName) != null) {
+                    Long value = (Long) avroRecord.get(fieldName);
+                    result = convertDateTimeToString(value);
+                  }
+                  record.put(fieldName.toLowerCase(), result);
+                } else {
+                  record.put(fieldName.toLowerCase(), avroRecord.get(fieldName));
+                }
+              } else if (avroTypeName.equalsIgnoreCase(AVRO_RECORD) && avroTypeSchema.getName().equalsIgnoreCase("scribe_header") && scribeHeaderRecord != null) {
+                record.put(fieldName.toLowerCase(), scribeHeaderRecord.get(fieldName));
+              } else if (avroTypeName.equalsIgnoreCase(AVRO_RECORD)) {
+                //convert fields inside nested object to parquet data
+                GenericRecord nestedRecord = (GenericRecord) avroRecord.get(fieldName);
+                Schema nestedRecordschema = schema.getField(fieldName.toLowerCase()).schema().isUnion() ? schema.getField(fieldName.toLowerCase()).schema().getTypes().get(1) : schema.getField(fieldName.toLowerCase()).schema();
+                record.put(fieldName.toLowerCase(), generateParquetStructuredAvroData(nestedRecordschema, nestedRecord));
+              } else {
+                // this is for primitive data types which do not need any conversion
+                record.put(fieldName.toLowerCase(), avroRecord.get(fieldName));
+              }
+            } catch (Exception e) {
+              LOG.error(String.format("Exception in converting avro field to parquet in ScribeParquetAvroConverter: Schema: %s, field: %s, typeName: %s, exception: %s", schema.getName(), fieldName, avroTypeName, e));
+            }
+          }
+        }
+      } catch (NullPointerException e){
```

Comment:
I would never catch NullPointerException. Please identify the case and handle it properly. This may mask other issues in your code.

Reply:
Also, if you are experiencing NullPointerException. It's most likely un recoverable error. You need to understand the implication of it, is it data loss, or something else you should be concerned about?